source/client-side-encryption/client-side-encryption.rst

```diff
@@ -591,6 +623,59 @@ the user. Refer:
 - `Detecting csfle Availability`_
 
+EncryptedFieldConfigMap
+^^^^^^^^^^^^^^^^^^^^^^^
+
+``EncryptedFieldConfigMap`` maps a collection namespace to an ``EncryptedFieldConfig``.
+
+``EncryptedFieldConfigMap`` only applies to FLE 2.
+
+If a collection is present on both the ``EncryptedFieldConfigMap`` and ``schemaMap``, libmongocrypt_ will error on initialization. See :ref:`fle2-and-fle1-error`.
+
+If a collection is present on the ``EncryptedFieldConfigMap``, the behavior of ``CreateCollection()`` and ``Collection.Drop()`` is altered. See :ref:`fle2-createcollection-drop`.
+
+Automatic encryption in FLE 2 is configured with the ``EncryptedFieldConfig``.
+
+If a collection is not present on the ``EncryptedFieldConfig`` a server-side collection ``EncryptedFieldConfig`` may be used by libmongocrypt_.
+Drivers MUST include the following in the documentation for MongoClient:
+
+   Supplying an ``EncryptedFieldConfigMap`` provides more security than relying on an ``EncryptedFieldConfig`` obtained from the server.
+   It protects against a malicious server advertising a false ``EncryptedFieldConfig``.
+
+.. _fle2-createcollection-drop:
+
+FLE 2 ``CreateCollection()`` and ``Collection.Drop()``
+------------------------------------------------------
+
+A collection supporting FLE 2 requires an index and three additional collections.
+
+A call to a driver helper ``CreateCollection(collectionName, collectionOptions)`` must check if the collection namespace (``<databaseName>.<collectionName>``) is in ``AutoEncryptionOpts.EncryptedFieldConfigMap``.
+If it is, then do the following:
+
+- Create the collection with name ``EncryptedFieldConfig["escCollection"]`` using default options.
+  If ``EncryptedFieldConfig["escCollection"]`` is not set, use the collection name ``enxcol_.<collectionName>.esc``.
+  Creating this collection MUST NOT check if the collection namespace is in the ``AutoEncryptionOpts.EncryptedFieldConfigMap``.
+- Create the collection with name ``EncryptedFieldConfig["eccCollection"]`` using default options.
+  If ``EncryptedFieldConfig["eccCollection"]`` is not set, use the collection name ``enxcol_.<collectionName>.ecc``.
+  Creating this collection MUST NOT check if the collection namespace is in the ``AutoEncryptionOpts.EncryptedFieldConfigMap``.
+- Create the collection with name ``EncryptedFieldConfig["ecocCollection"]`` using default options.
+  If ``EncryptedFieldConfig["ecocCollection"]`` is not set, use the collection name ``enxcol_.<collectionName>.ecoc``.
+  Creating this collection MUST NOT check if the collection namespace is in the ``AutoEncryptionOpts.EncryptedFieldConfigMap``.
+- Create the collection ``collectionName`` with ``collectionOptions`` and the option ``encryptedFields`` set to the ``EncryptedFieldConfig``.
+- Create the the index ``{"__safeContent__": 1}`` on collection ``collectionName``.
+
+Drivers MUST NOT add a public option to directly set the ``encryptedFields`` option in the ``CreateCollection`` helper.
```

Comment:
Right now, at least the Node.js driver just passes through collection creation options, i.e. I can write `await db.createCollection('foo', { encryptedFields: { fields: [{ path: 'field', bsonType: 'string', keyId: UUID('00000000-0000-0000-0000-000000000000') }] } })` and that will work. Should the driver error out in this case?

Reply:
Good point. The rationale for no a public `encryptedFields` option is to prevent users from creating a collection with `encryptedFields` without the index or state collections.

IMO we should not make it a driver error if `encryptedFields` is set in `RunCommand`. If some drivers pass along options through `createCollection`, this seems like a similar case.

I have removed this sentence. I want drivers to not add an `encryptedFields` option on their `CreateCollectionOptions` type if they have it. But drivers that are passing options along (Node, and also C) should not reject it.